File: experiments/runs/2026-09-17_00-24-43_claim-office-example-mapping_exact-ptdd-v1.1-refactor-subagent-cc_opus-5-no-thinking-2/src/policy-cover.ts
import { insuranceValueOf } from "./item-price-rules.js";
import type { Item } from "./policy.js";

/** The policy's insurance sum is the sum of its items' insurance values. */
export function insuranceSum(items: readonly Item[]): number {
  return items.reduce((sum, item) => sum + insuranceValueOf(item), 0);
}

/** The total payout per policy is capped at twice the insurance sum. */
const PAYOUT_CAP_MULTIPLE = 2;

export function payoutCap(items: readonly Item[]): number {
  return insuranceSum(items) * PAYOUT_CAP_MULTIPLE;
}

File: experiments/runs/2026-09-20_11-57-09_claim-office-java-example-mapping_exact-ptdd-v1.1-refactor-subagent-cc_opus-5-no-thinking/src/main/java/PolicyCover.java
import java.util.List;

/**
 * What a policy is worth to the office: the insurance sum it has underwritten, and the
 * cap on everything it will ever pay out on that policy.
 *
 * The insurance sum is the plain sum of the listed insurance values of the covered items.
 * It is deliberately taken from the price list rather than from the premium: a block
 * discount lowers what the customer pays without lowering what the office insures, and a
 * risk surcharge raises the premium without raising the cap.
 */
public class PolicyCover {

    private static final int CAP_MULTIPLE_OF_INSURANCE_SUM = 2;

    private final int insuranceSum;

    private PolicyCover(int insuranceSum) {
        this.insuranceSum = insuranceSum;
    }

    public static PolicyCover of(List<Item> items) {
        PriceList priceList = new PriceList();
        int sum = 0;
        for (Item item : items) {
            sum += priceList.insuranceValueOf(item);
        }
        return new PolicyCover(sum);
    }

    /**
     * The ceiling the office underwrote: everything it will ever pay out on this policy,
     * twice the insurance sum.
     */
    public int cap() {
        return CAP_MULTIPLE_OF_INSURANCE_SUM * insuranceSum;
    }
}
